src/Model/Behavior/SearchBehavior.php

```diff
@@ -62,10 +62,18 @@ public function findSearch(Query $query, array $options)
                 'to be nested under key "search" in find() options.'
             );
         }
-
+        
         $filters = $this->_getAllFilters(Hash::get($options, 'collection', 'default'));
 
-        $params = $this->_flattenParams((array)$options['search']);
+        $flattenWhitelist = [];
+        foreach($filters as $filter) {
```

Comment:
Ah good points, thanks. I will take care of this in a week as I'm traveling at the moment.